litestar/middleware/_internal/exceptions/middleware.py

```diff
@@ -146,9 +147,19 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         Returns:
             None
         """
+        scope_state = ScopeState.from_scope(scope)
+
+        async def capture_response_started(event: Message) -> None:
+            if event["type"] == "http.response.start":
+                scope_state.response_started = True
+            await send(event)
+
         try:
-            await self.app(scope, receive, send)
+            await self.app(scope, receive, capture_response_started)
         except Exception as e:  # noqa: BLE001
+            if scope_state.response_started:
+                raise
```

Comment:
Should we raise this from our own exception, e.g.,

```suggestion
                raise LitestarException("Exception caught after response started") from e
```

Reply:
Yeah, I'd say so. The extra context here could be really useful when debugging